### PR TITLE
PostgreSQL: Fix tls certificate issue by downgrading lib/pq

### LINKED ISSUE #65816: postgres: tls-mode `verify-ca` does not work correctly (behaves like `verify-full`)
summary by @gabor  : the postgres TLS mode `verify-ca` does not work correctly, and behaves the same as `verify-full`. to reproduce:  try this scenario: https://github.com/grafana/oss-big-tent-tools/tree/main/tls-setups/postgres#verify-server-cert-ignore-host . it should work, but it does not.

---

**What happened**:
Grafana Versions >= 9.4.0 cause the following issues in Panels that use an PostgreSQL (AWS RDS) as Datasource and
the following TLS settings:

**Issues**

```
db query error: x509: certificate is valid for <DB instance url>, not <route53 url to db>
```
```
db query error: x509: certificate is not valid for any ....
```

**Settings**

Host: "insert AWS RDS Postgres Host route53"
Database: "insert DB name"
User+Password
TLS/SSL Mode: *verify-ca*
TLS/SSL Method: *Certificate content*
TLS/SSL Root Certificate: "insert official aws rds certificate"

PostgreSQL details
Version: *11*

*One*(the first) certificate out of the official AWS RDS root certificates eu-central-1 bundle is used:
https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.SSL.html
*[eu-central-1-bundle.pem](https://truststore.pki.rds.amazonaws.com/eu-central-1/eu-central-1-bundle.pem)*

**What you expected to happen**:
The panels should work in the same way they do in Grafana v9.3.8

**How to reproduce it (as minimally and precisely as possible)**:
1. Use Grafana v9.3.8
2. Add an AWS RDS Postgres Database as Datasource incl. the TLS settings above
3. use a route53 to access the AWS RDS
4. it should work
5. upgrade Grafana to a Version >=9.4.0
6. the issues mentioned above should appear and the panels dont work

**Anything else we need to know?**:

**Environment**:
- Grafana version: >= 9.4.0
- Data source type & version: PostgreSQL
- OS Grafana is installed on: [Grafana official image](https://hub.docker.com/r/grafana/grafana) deployed as Stateful Set in Openshift

### PR DESCRIPTION
**Pull request by zoltanbedi**
This PR tries to fix the issue stated in #65816 by downgrading the lib/pq library.

**Which issue(s) does this PR fix?**:

Fixes #65816

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -73,7 +73,7 @@ require (
 	github.com/jmespath/go-jmespath v0.4.0
 	github.com/json-iterator/go v1.1.12
 	github.com/jung-kurt/gofpdf v1.16.2
-	github.com/lib/pq v1.10.7
+	github.com/lib/pq v1.10.6
```
Comment (papagian) [inline go.mod:76]: I would suggest adding a comment to prevent future attempts to upgrade until this is fixed

**Review by papagian — APPROVED** [review]
given the [diff](https://github.com/lib/pq/compare/v1.10.6...v1.10.7) between the two versions
I think it's safe to downgrade it

**Review by sakjur — APPROVED** [review]
LGTM :+1: I don't see any harm in downgrading based on looking at https://github.com/lib/pq/compare/v1.10.6...v1.10.9 and https://github.com/lib/pq/compare/v1.10.6...v1.10.7

File: go.mod (hunk 1 above)
Comment (zoltanbedi) [reply go.mod:76]: Thanks @papagian added a comment at the bottom. Is that okay?

**Pull request merged**